hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -1473,42 +1474,55 @@ public boolean setReplication(Path src, short replication)
   /**
    * Renames Path src to Path dst
    * <ul>
-   *   <li>Fails if src is a file and dst is a directory.</li>
-   *   <li>Fails if src is a directory and dst is a file.</li>
-   *   <li>Fails if the parent of dst does not exist or is a file.</li>
+   *   <li>Fails if src is a file and dest is a directory.</li>
+   *   <li>Fails if src is a directory and dest is a file.</li>
+   *   <li>Fails if the parent of dest does not exist or is a file.</li>
    * </ul>
    * <p>
    * If OVERWRITE option is not passed as an argument, rename fails
-   * if the dst already exists.
+   * if the dest already exists.
    * <p>
    * If OVERWRITE option is passed as an argument, rename overwrites
-   * the dst if it is a file or an empty directory. Rename fails if dst is
+   * the dest if it is a file or an empty directory. Rename fails if dest is
    * a non-empty directory.
    * <p>
    * Note that atomicity of rename is dependent on the file system
    * implementation. Please refer to the file system documentation for
    * details. This default implementation is non atomic.
-   * <p>
-   * This method is deprecated since it is a temporary method added to
-   * support the transition from FileSystem to FileContext for user
-   * applications.
    *
-   * @param src path to be renamed
-   * @param dst new path after rename
-   * @throws FileNotFoundException src path does not exist, or the parent
-   * path of dst does not exist.
+   * @param source path to be renamed
+   * @param dest new path after rename
+   * @throws FileNotFoundException source path does not exist, or the parent
+   * path of dest does not exist.
    * @throws FileAlreadyExistsException dest path exists and is a file
    * @throws ParentNotDirectoryException if the parent path of dest is not
    * a directory
    * @throws IOException on failure
    */
-  @Deprecated
```

Comment:
You remove the deprecaiton here from the abstract FS. How will it affect the target release it can get in?
FS is 
```@InterfaceAudience.Public
@InterfaceStability.Stable```

Reply:
Also it changed to `protected` to `public`. The same question applies.

Reply:
that's exactly the purpose: take a method which is protected and used only by FileContext, and make it public. It's been in for a while, been stable -and now needs tests and documentation of what it really does